File: frontend/src/config/edition.ts
/**
 * Edition management for Elements Platform frontend
 *
 * Controls which modules and features are available based on the edition.
 * Edition is set at build time via VITE_EDITION environment variable.
 */

export enum Edition {
  CORE = 'core',
  ENTERPRISE = 'enterprise'
}

// Current edition - set via environment variable at build time
export const CURRENT_EDITION = (import.meta.env.VITE_EDITION as Edition) || Edition.CORE;

// Module availability mapping for each edition
export const EDITION_MODULES: Record<Edition, string[]> = {
  [Edition.CORE]: ['portal', 'hr', 'it'],
  [Edition.ENTERPRISE]: ['portal', 'hr', 'it', 'tasks', 'knowledge_core']
};

// Feature availability mapping for each edition
export const EDITION_FEATURES: Record<Edition, Record<string, boolean>> = {
  [Edition.CORE]: {
    knowledgeBase: false,
    tasks: false,
    rocketchat: false,
    zabbix: false,
    qdrant: false,
    llm: false
  },
  [Edition.ENTERPRISE]: {
    knowledgeBase: true,
    tasks: true,
    rocketchat: true,
    zabbix: true,
    qdrant: true,
    llm: true
  }
};

/**
 * Checks if a module is available in the current edition
 * @param module Module name (e.g., "tasks", "knowledge_core")
 * @returns true if module is available, false otherwise
 */
export function isModuleAvailable(module: string): boolean {
  return EDITION_MODULES[CURRENT_EDITION].includes(module);
}

/**
 * Checks if a feature is available in the current edition
 * @param feature Feature name (e.g., "rocketchat", "zabbix")
 * @returns true if feature is available, false otherwise
 */
export function isFeatureAvailable(feature: keyof typeof EDITION_FEATURES[Edition.CORE]): boolean {
  return EDITION_FEATURES[CURRENT_EDITION][feature] || false;
}

/**
 * Gets the human-readable edition name
 * @returns Edition name (e.g., "Core", "Enterprise")
 */
export function getEditionName(): string {
  return CURRENT_EDITION === Edition.ENTERPRISE ? 'Enterprise' : 'Core';
}

/**
 * Gets all available modules for the current edition
 * @returns Array of module names
 */
export function getAvailableModules(): string[] {
  return EDITION_MODULES[CURRENT_EDITION];
}

File: backend/core/edition.py
"""
Edition management for Elements Platform.

Defines available editions (Core and Enterprise) and controls
which modules and integrations are available in each edition.
"""

from enum import Enum
from typing import List, Dict


class Edition(str, Enum):
    """Available product editions."""
    CORE = "core"
    ENTERPRISE = "enterprise"


# Current edition - set at build time via Dockerfile
# This value is replaced during Docker image build
CURRENT_EDITION = Edition.CORE


# Module availability mapping for each edition
EDITION_MODULES: Dict[Edition, List[str]] = {
    Edition.CORE: ["portal", "hr", "it"],
    Edition.ENTERPRISE: ["portal", "hr", "it", "tasks", "knowledge_core"]
}


# Integration availability mapping for IT module
EDITION_INTEGRATIONS: Dict[Edition, List[str]] = {
    Edition.CORE: ["email", "telegram", "ldap"],
    Edition.ENTERPRISE: ["email", "telegram", "ldap", "rocketchat", "zabbix"]
}


def get_allowed_modules() -> List[str]:
    """
    Returns the list of modules allowed in the current edition.

    Returns:
        List of module names (e.g., ["portal", "hr", "it"])
    """
    return EDITION_MODULES[CURRENT_EDITION]


def get_allowed_integrations() -> List[str]:
    """
    Returns the list of integrations allowed in the current edition.

    Returns:
        List of integration names (e.g., ["email", "telegram"])
    """
    return EDITION_INTEGRATIONS[CURRENT_EDITION]


def is_module_allowed(module: str) -> bool:
    """
    Checks if a module is allowed in the current edition.

    Args:
        module: Module name to check (e.g., "tasks")

    Returns:
        True if the module is available in current edition, False otherwise
    """
    return module in EDITION_MODULES[CURRENT_EDITION]


def is_integration_allowed(integration: str) -> bool:
    """
    Checks if an integration is allowed in the current edition.

    Args:
        integration: Integration name to check (e.g., "rocketchat")

    Returns:
        True if the integration is available in current edition, False otherwise
    """
    return integration in EDITION_INTEGRATIONS[CURRENT_EDITION]


def get_edition_name() -> str:
    """
    Returns the human-readable name of the current edition.

    Returns:
        Edition name (e.g., "Core", "Enterprise")
    """
    return "Enterprise" if CURRENT_EDITION == Edition.ENTERPRISE else "Core"
